### ISSUE TITLE
Error reporter in SNAFU

### ISSUE THREAD

**Issue by YjyJeff** [shepmaster/snafu#354]
According to the [Error handling project](https://blog.rust-lang.org/inside-rust/2021/07/01/What-the-error-handling-project-group-is-working-towards.html#2-error-reporter), they recommend using the error reporter to report the error and 

> An error type with a source error should either return that error via source or include that source's error message in its own Display output, but never both.

Does SNAFU have the error reporter that can print the Error in multiple lines like anyhow ? 

```rust
// Outputs
// Error: outermost error: second error: root error
//
// Caused by:
//    0: second error: root error
//    1: root error
```

**Comment by shepmaster** [comment]
I happen to be working on that at the moment (#355) so it should be available in SNAFU 0.7.2 🤞. Please feel free to skim the docs in that PR and provide feedback.

**Comment by YjyJeff** [comment]
Looking forward to it!
